src/MCPClient/README.md

```diff
@@ -2,11 +2,80 @@
-`MCPServer`. These tasks are dispatched via Gearman. Technically speaking,
-`MCPClient` is a pool of Gearman workers each one running in a separate thread.
-
-The programs responsible of performing the Gearman jobs are contained in the
-`lib/clientScripts/` directory. We frequently implement them as Python or Bash
-scripts. When using Python, it's preferred to use the `logging` module - that
-gives us the possibility to dispatch the log messages to more than one handler,
-e.g. `stdout`, `stderr`, etc...
+`MCPServer`. These tasks are dispatched via Gearman.
+
+The programs responsible for performing the Gearman jobs are contained in the
+`lib/clientScripts/` directory. Each program is a Python module that
+implements the following function:
+
+    def call(jobs)
+
+The `jobs` parameter is an array of `Job` objects, each corresponding
+to a task to be performed. If there are arguments associated with the
+task, `job.args` will hold those. When each job is run, it is
+expected to produce an exit code along with output and error
+streams. You set the exit code for a job with:
+
+    job.set_status(int)
+
+To record standard output and standard error, there are several
+methods: write_* for storing literal strings; print_* for storing
+literal strings with newlines added; pyprint for a Python
+`print`-compatible API. See the Job class for more information on
+these.
+
+There are numerous examples of client scripts in the `clientScripts`
+directory. Commonly they will follow a pattern like:
+
+    from custom_handlers import get_script_logger
+    # ...
+    logger = get_script_logger("archivematica.mcp.client.myModuleName")
+
+    # ...
+
+    def call(jobs):
+        with transaction.atomic():
+            for job in jobs:
+                with job.JobContext(logger=logger):
+                    result = process_job(job.args)
+                    job.set_status(result)
+
+Some notable features:
+
+  * Where the task will insert/update/delete rows from the database,
+    we wrap it in a transaction. Besides providing atomicity if
+    something goes wrong, this also boosts performance significantly.
+
+  * Generally we iterate over each job one-by-one, although nothing
+    stops you from working in larger batches if that suits.
+
+  * `JobContext` is a convenience class that executes its body in a
+    modified context:
+
+      - Any uncaught exceptions will be logged as standard error
+        output, and the job's status will be set to 1.
+
+      - If you supply the `logger` keyword, your global logger will be
+        configured to send its output to the job's standard error
+        stream.
+
+## Concurrency
+
+Some tasks benefit from running multiple processes at once--giving
+better throughput. To instruct MCPClient to run your task in multiple
+concurrent processes, provide a function in your module as follows:
+
+    import multiprocessing
+    def concurrent_instances():
+        return multiprocessing.cpu_count()
```

Comment:
Instead of doing cpu_count() here, can you have this return the value of `ARCHIVEMATICA_MCPCLIENT_MCPCLIENT_NUMBEROFTASKS`
See https://github.com/artefactual/archivematica/tree/stable/1.7.x/src/MCPClient/install#environment-variables

This environment variable often gets set to the cpu count during deployment 
https://github.com/artefactual/archivematica/blob/b6dcfb07a6be5957a5085efd1fecd8462fdc3a91/src/MCPClient/lib/settings/common.py#L81

archivematicaClient already detects the number of cpu cores (see https://github.com/artefactual/archivematica/blob/051ded0f3e079f43594e0c90e862cc4db33ebd93/src/MCPClient/lib/archivematicaClient.py#L187 )

Reply:
Ah, it looks like this variable got the chop in 6c33d34c2a8eed51e216a8c0d57bff343dbc5dad.  I don't have strong feelings about where we get this value from, as long as we know that it's at least 1 :)  Thoughts, anyone?

Reply:
The purpose of `NUMBER_OF_TASKS` wasn't exactly the same although very similar but I think it'd be fine if it needs to be reintroduced. This could also be done after we merge this PR, that would be at least my preference. I think it's going to be hard to keep this work up to date, conflicts arise quickly because it touches many files.

Reply:
That sounds good to me.  We can leave `NUMBER_OF_TASKS` out, and consider adding it back in after, if we need it.  We may not need to.